celltype_annotation/analysis/cell-assign-sarcoma.Rmd

```diff
@@ -0,0 +1,435 @@
+  tidyr::drop_na(cellontology_id, Symbol)
+
+# what types of cells are represented here? 
+unique(muscle_markers_df$cancer_type)
+unique(muscle_markers_df$cell_name)
+```
+
+
+```{r}
+# create a binary matrix of genes by cell type markers 
+cellmarker_mtx <- muscle_markers_df |>
+  dplyr::select(cellontology_id, Symbol) |> 
+  tidyr::pivot_wider(Symbol,
+                     names_from = cellontology_id,
+                     values_from = cellontology_id,
+                     values_fn = list(cellontology_id = length),
+                     values_fill = list(cellontology_id = 0)) |> 
+  tibble::column_to_rownames("Symbol") |>
+  dplyr::mutate(other = 0)
+
+# replace length with 1
+cellmarker_mtx[cellmarker_mtx > 1] <- 1
+```
+
+
+
+```{r}
+# convert rownames to ensembl ids
+rowdata_df <- rowData(processed_sce) |>
+  as.data.frame() |>
+  tibble::rownames_to_column("ensembl_id") |>
+  dplyr::select(ensembl_id, gene_symbol)
+
+# merge reference with rowdata and replace gene symbols with ensembl
+cellmarker_mtx <- cellmarker_mtx |>
+  tibble::rownames_to_column("gene_symbol") |>
+  dplyr::left_join(rowdata_df) |>
+  # drop any rows where there is no ensembl id
+  tidyr::drop_na(ensembl_id) |>
+  dplyr::select(-gene_symbol) |>
+  dplyr::relocate(ensembl_id)
+```
+
+```{r}
+# export mtx 
+readr::write_csv(cellmarker_mtx, file.path("..", "references", "cell_marker_muscle_mtx.csv"))
+```
+
+```sh
+# run cell assign with muscle ref
+#python ../scripts/cell-assign.py \
+#  --input_anndata "../data/anndata/SCPCL000488_anndata.hdf5" \
+#  --output_predictions "../data/anndata/SCPCL000488_muscle.tsv" \
+#  --reference "../references/cell_marker_muscle_mtx.csv"
+```
+
+
+```{r}
+# read in prediction scores 
+cellmarker_predictions <- readr::read_tsv(file.path(anndata_dir, "SCPCL000488_muscle.tsv"))
+```
+
+```{r}
+# grab cell label names corresponding to cell ontology ids
+cl_ont <- ontoProc::getCellOnto()
+
+# replace column names(ontology ids) with cell label
+colnames(cellmarker_predictions) <- stringr::str_replace(colnames(cellmarker_predictions), "_", ":")
+cl_ids <- colnames(cellmarker_predictions)[-grep("other|barcode", colnames(cellmarker_predictions))]
+colnames(cellmarker_predictions)[-grep("other|barcode", colnames(cellmarker_predictions))] <- cl_ont$name[cl_ids]
+```
+
+
+
+```{r}
+# get assignments for each cell from prediction matrix 
+cellmarker_assignments <- get_celltype_assignments(cellmarker_predictions)
+
+# print table of assignments
+cellmarker_assignments |>
+  dplyr::count(celltype) |>
+  dplyr::arrange(desc(n))
+  
+```
+
+```{r}
+# heatmap of prediction scores 
+cellmarker_predictions |>
+  dplyr::select(-barcode) |> 
+  as.matrix() |> 
+  pheatmap::pheatmap(show_rownames = FALSE)
+```
+
+Here most of the cells appear to be assigned to macrophages... 
+According to the known cell type annotations most of the cell types here are tumor cells and of those tumor cells they are mostly myoblasts. 
```

Comment:
Nothing appears to be assigned as a cancer type: are we sure they made it into the reference?
(the table I suggested above might be enhanced by translating the cell ontology labels)